lib/horde/dynamic_supervisor_impl.ex

```diff
@@ -58,6 +58,11 @@ defmodule Horde.DynamicSupervisorImpl do
 
   def handle_continue({:set_members, nil}, state), do: {:noreply, state}
 
+  def handle_continue({:set_members, :auto}, state) do
+    Horde.NodeListener.initial_set(state.name)
```

Comment:
I'm unsure about this bit right here. This will result in a race condition. It's possible that some requests will come in to this supervisor before it's had a chance to set its members.

Reply:
Ah, good catch. I've fixed it to it a synchronous part of the supervisor startup (and also refactored the `Horde.Registry` initialisation to do the same, if only for consistency).